paddle/phi/infermeta/unary.cc

```diff
@@ -768,6 +768,16 @@ void DirichletInferMeta(const MetaTensor& alpha, MetaTensor* out) {
   out->set_dtype(alpha.dtype());
 }
 
+void DistReduceInferMeta(const MetaTensor& x, MetaTensor* out) {
```

Comment:
It's better to use the same style to name `AllReduceInferMeta, DistReduceInferMeta, DistBroadcastMeta` and so on.

Reply:
> It's better to use the same style to name `AllReduceInferMeta, DistReduceInferMeta, DistBroadcastMeta` and so on.

This conflict was encountered later and requires a thorough framework resolution, so it is not fully unified. we can unifiy it later